### PR TITLE
[vocab/extractor] use pickle to dump and load Vocabulary and Extractor

### PR DESCRIPTION
**Pull request by Jiaqiang-Ruan**
### Description of changes
use `__getstate__` and `__setstate__` to save vocabulary instead of `state` and `from_state` function.

### Possible influences of this PR.
The way to save vocabulary will be changed. After this PR, users can use pickle to save vocabulary directly.

### Test Conducted
Pickle load and dump is tested.

### CODE REVIEW

File: forte/data/vocabulary.py [hunk 1]
```diff
@@ -116,13 +116,14 @@ def __init__(self, method: str, need_pad: bool, use_unk: bool):
...
     def __getstate__(self):
         r"""For serialization."""
-        state = {}
-        state["method"] = self.method
-        state["need_pad"] = self.need_pad
-        state["use_unk"] = self.use_unk
-        state["element2id_dict"] = self.element2id_dict
-        state["id2element_dict"] = self.id2element_dict
-        state["next_id"] = self.next_id
+        state = {
```
Comment (hunterhector) [inline forte/data/vocabulary.py:119]: Is this function simply return everything of this vocab class into a dict? If so, you don't need to implement __getstate__ and __setstate__ at all. Python's default `__dict__` will be used (https://docs.python.org/3/library/pickle.html#object.__getstate__)

File: forte/data/extractor/base_extractor.py [hunk 2]
```diff
@@ -119,31 +119,18 @@ def default_configs():
...
-        return {
-            "vocab_method": self.config.vocab_method,
-            "need_pad": self.config.need_pad,
-            "vocab_use_unk": self.config.vocab_use_unk,
-            "entry_type": self.config.entry_type,
-            "vocab": self.vocab.state if self.vocab else None,
+    def __getstate__(self):
+        r"""For serializaion."""
+        state = {
+            "config": self.config,
```
Comment (hunterhector) [inline forte/data/extractor/base_extractor.py:125]: Similar to the other comment, if your serialization step actually saves all class fields, without special treatment, you don't need to implement __getstate__ and __setstate__

**Review by hunterhector — COMMENTED** [review]
I think these classes can be directly pickled

File: forte/data/extractor/base_extractor.py (hunk 2 above)
Comment (Jiaqiang-Ruan) [reply forte/data/extractor/base_extractor.py]: Got it.

File: forte/data/vocabulary.py (hunk 1 above)
Comment (Jiaqiang-Ruan) [reply forte/data/vocabulary.py]: That's good to know. Updated.

**Review by hunterhector — APPROVED** [review]
should be find

**Pull request merged**